### DISCUSSION TITLE
MaxMind Database as having issues with RIPE and IPDENY

### DISCUSSION THREAD

**Discussion by tricky1024** [friendly-bits/geoip-shell#21] [General]
Would it be possible to have maxmind used as a source as im having an issue in the UK with IP address ranges missing from RIPE and IPDENY.

**Comment by friendly-bits** [comment]
Hi,  do you know for a fact that Maxmind's database includes these ranges for UK? If so then which one - GeoLite2 (the free one) or GeoIP2 (the paid one)?

**Reply by tricky1024** [reply]
So its in the database and I can check via: https://www.maxmind.com/en/geoip-demo but free or paid I'm not sure, the range that is in the MaxMind DB is: 149.102.62.0/23 doesn't seem to appear in RIPE or IPDENY

**Reply by friendly-bits** [reply]
Assuming the ranges are included in the paid database, will you actually pay for it? I may consider implementing this but first I need to understand whether there is an actual use case.

**Reply by tricky1024** [reply]
I run the free database in other places so let me check and see if i can check against it.

**Reply by tricky1024** [reply]
i have tested maxmind's free database with that same IP and it works perfectly, i really feel integrating the free database would be useful.

**Reply by friendly-bits** [reply]
Thanks for the suggestion and for testing. I'll look into implementing this feature but it may take some time because I'm currently working on implementation of another fairly big feature which is outbound geoblocking. When that is complete, I'll be open to implementing additional features.

**Reply by tricky1024** [reply]
perfect, thank you so much

**Reply by friendly-bits** [reply]
Hi @tricky1024 , I just released geoip-shell v0.6.4 which adds support for MaxMind. Please let me know how it works for you.

**Reply by tricky1024** [reply]
many thanks indeed, ill get that installed and test this week.
